### ISSUE TITLE
typos: "an function"

### ISSUE THREAD

**Issue by dilyanpalauzov** [saltstack/salt#46558]
```diff
diff --git a/doc/man/salt-cloud.1 b/doc/man/salt-cloud.1
--- a/doc/man/salt-cloud.1
+++ b/doc/man/salt-cloud.1
@@ -93,7 +93,7 @@ argument requires one or more instance names to be specified.
 .INDENT 0.0
 .TP
 .B \-f <FUNC\-NAME> <PROVIDER>, \-\-function=<FUNC\-NAME> <PROVIDER>
-Perform an function that may be specific to this cloud provider, that does
+Perform a function that may be specific to this cloud provider, that does
 not apply to an instance. This argument requires a provider to be specified
 (i.e.: nova).
 .UNINDENT
diff --git a/doc/man/salt.7 b/doc/man/salt.7
--- a/doc/man/salt.7
+++ b/doc/man/salt.7
@@ -52707,7 +52707,7 @@ argument requires one or more instance names to be specified.
 .INDENT 0.0
 .TP
 .B \-f <FUNC\-NAME> <PROVIDER>, \-\-function=<FUNC\-NAME> <PROVIDER>
-Perform an function that may be specific to this cloud provider, that does
+Perform a function that may be specific to this cloud provider, that does
 not apply to an instance. This argument requires a provider to be specified
 (i.e.: nova).
 .UNINDENT
diff --git a/doc/ref/cli/salt-cloud.rst b/doc/ref/cli/salt-cloud.rst
--- a/doc/ref/cli/salt-cloud.rst
+++ b/doc/ref/cli/salt-cloud.rst
@@ -46,7 +46,7 @@ Execution Options
 
 .. option:: -f <FUNC-NAME> <PROVIDER>, --function=<FUNC-NAME> <PROVIDER>
 
-    Perform an function that may be specific to this cloud provider, that does
+    Perform a function that may be specific to this cloud provider, that does
     not apply to an instance. This argument requires a provider to be specified
     (i.e.: nova).
 
diff --git a/doc/topics/pillar/index.rst b/doc/topics/pillar/index.rst
--- a/doc/topics/pillar/index.rst
+++ b/doc/topics/pillar/index.rst
@@ -409,7 +409,7 @@ module. This module includes several functions, each of them with their own
 use. These functions include:
 
 - :py:func:`pillar.item <salt.modules.pillar.item>` - Retrieves the value of
-  one or more keys from the :ref:`in-memory pillar datj <pillar-in-memory>`.
+  one or more keys from the :ref:`in-memory pillar data <pillar-in-memory>`.
 - :py:func:`pillar.items <salt.modules.pillar.items>` - Compiles a fresh pillar
   dictionary and returns it, leaving the :ref:`in-memory pillar data
   <pillar-in-memory>` untouched. If pillar keys are passed to this function
diff --git a/salt/utils/parsers.py b/salt/utils/parsers.py
--- a/salt/utils/parsers.py
+++ b/salt/utils/parsers.py
@@ -1411,7 +1411,7 @@ class ExecutionOptionsMixIn(six.with_metaclass(MixInMeta, object)):
             nargs=2,
             default=None,
             metavar='<FUNC-NAME> <PROVIDER>',
-            help='Perform an function that may be specific to this cloud '
+            help='Perform a function that may be specific to this cloud '
                  'provider, that does not apply to an instance. This '
                  'argument requires a provider to be specified (i.e.: nova).'
         )
```

**Comment by garethgreenaway** [comment]
@dilyanpalauzov Thanks for the report.  Can you provide some version information where you're seeing these typos?  Thanks!

**Comment by dilyanpalauzov** [comment]
These are on the git `master` branch.

**Comment by rallytime** [comment]
@dilyanpalauzov The man pages are generated automatically before each release. They are based on the documentation. If you like to submit a change to update the docs, the man pages will pick up the changes next time they're generated. We'd love to get these cleaned up if you'd like to help! :)

**Comment by dilyanpalauzov** [comment]
The cited by me files are managed in git.  Please replace all occurrences of "an function" with "a function" and "datj" with "data".  The code for the man pages comes apparently from salt/utils/parsers.py, the snippet above suggests changing it, too.

**Comment by rallytime** [comment]
@dilyanpalauzov The `an function` references have been fixed with #46585 and the `datj` typo has been fixed with #46610.

Thanks for letting us know. :)
